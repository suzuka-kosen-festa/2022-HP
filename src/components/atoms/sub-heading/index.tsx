import tw from "twin.macro";

const SubHeading = tw.h2`font-zen text-text font-bold text-base sm:text-2xl select-none`;

export { SubHeading };
