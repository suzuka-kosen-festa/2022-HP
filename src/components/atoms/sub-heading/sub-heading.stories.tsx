import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { SubHeading } from ".";

type T = typeof SubHeading;
type Story = ComponentStoryObj<T>;

const data = {
  sentence: "This is test",
};

export default {
  args: {children: data.sentence},
  argTypes: {
    children: {description: "Context of sub heading", control: {type: "text"}},
  },
  component: SubHeading,
} as ComponentMeta<T>

export const Default: Story = {}
