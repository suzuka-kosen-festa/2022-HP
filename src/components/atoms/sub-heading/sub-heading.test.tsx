import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./sub-heading.stories";

const { Default } = composeStories(stories);

const options = {
  name: "This is test",
};

describe("(components) atoms/text", () => {
  test("to be atoms", () => {
    const { container } = render(<Default />);
    expect(container).toBeAtom();
  });
  test("to be [role=heading]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("heading", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
