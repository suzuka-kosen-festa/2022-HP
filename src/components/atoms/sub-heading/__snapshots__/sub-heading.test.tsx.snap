// Vitest Snapshot v1

exports[`(components) atoms/sub-heading > take snap shot 1`] = `
<div>
  <h2
    class="css-1a2c2qr"
  >
    This is test
  </h2>
</div>
`;
